manifests/init.pp

```diff
@@ -820,8 +820,77 @@
                 ]
               }
             }
+            '16.04': {
+              $default_pam_d_login_template = 'pam/login.ubuntu16.erb'
+              $default_pam_d_sshd_template  = 'pam/sshd.ubuntu16.erb'
+              $default_package_name         = 'libpam0g'
+
+              if $ensure_vas == 'present' {
+                $default_pam_auth_lines = [
```

Comment:
Ok, you're right, this is untested code. I just copied it from 14.04, given that the changes in the rest of the flies are minimum between releases. Are the values from 14.04 just the defaults  after installing pam_vas? Would it be enough to just install pam_vas under 16.04 and get the defaults into the manifest? Otherwise, probably the best option is as you suggested just to fail()